core/game_object.lua

```diff
@@ -2419,165 +2423,237 @@ function loadAPIs()
+            G.ASSET_ATLAS[v][name] = {image_data = G.ASSET_ATLAS[v].image_data:clone()}
+            if #SMODS.AltTextures[type][name].old_colours > 0 then 
+                G.ASSET_ATLAS[v][name].image_data:mapPixel(G.FUNCS.recolour_image)
+            end
+            G.ASSET_ATLAS[v][name].image = love.graphics.newImage(G.ASSET_ATLAS[v][name].image_data, {mipmaps = true, dpiscale = G.SETTINGS.GRAPHICS.texture_scaling})
         end
     end
 
-    function SMODS.Palette:create_colours(type, base_colour, alternate_colour)
-        if SMODS.ConsumableTypes[type].generate_colours then
-            return SMODS.ConsumableTypes[type]:generate_colours(HEX_HSL(base_colour),
-                alternate_colour and HEX_HSL(alternate_colour))
+    function SMODS.AltTexture:create_new_atlas()
+        SMODS.AltTextures[self.type][self.name].old_colours = {}
+        SMODS.AltTextures[self.type][self.name].new_colours = {}
+        SMODS.AltTexture:create_atlas(self.type, self.name)
+        local atlas_key
+        if self.type == "Suit" then
+            atlas_key = "cards_1"
+            G.ASSET_ATLAS["ui_1"][self.name] = {image_data = G.ASSET_ATLAS["ui_1"]["Default"].image_data:clone()}
+            G.PALETTE = {
+                old_colours = {HEX("235955"),HEX("3c4368"),HEX("f06b3f"),HEX("f03464")},
+                new_colours = {HEX(SMODS.AltTextures[self.type][self.name].suits.Clubs), HEX(SMODS.AltTextures[self.type][self.name].suits.Spades), HEX(SMODS.AltTextures[self.type][self.name].suits.Diamonds), HEX(SMODS.AltTextures[self.type][self.name].suits.Hearts)}
+            }
+            G.ASSET_ATLAS["ui_1"][self.name].image_data:mapPixel(G.FUNCS.recolour_image)
+            G.ASSET_ATLAS["ui_1"][self.name].image = love.graphics.newImage(G.ASSET_ATLAS["ui_1"][self.name].image_data, {mipmaps = true, dpiscale = G.SETTINGS.GRAPHICS.texture_scaling})
+        elseif self.type == "Spectral" then
+            atlas_key = "Spectral"
+            if G.ASSET_ATLAS[self.atlas_key.."_e"] then
+                G.ASSET_ATLAS["soul"][self.name] = G.ASSET_ATLAS[self.atlas_key.."_e"]
+            else
+                G.ASSET_ATLAS["soul"][self.name] = {image_data = G.ASSET_ATLAS["soul"].image_data:clone()}
+                G.ASSET_ATLAS["soul"][self.name].image = love.graphics.newImage(G.ASSET_ATLAS["soul"][self.name].image_data, {mipmaps = true, dpiscale = G.SETTINGS.GRAPHICS.texture_scaling})
+            end
+        elseif self.type == "Seal" then
+            atlas_key = "centers"
+        elseif self.type == "Tag" then
+            atlas_key = "tags"
+        else
+            atlas_key = self.type
         end
-        return { HEX(base_colour) }
+        G.ASSET_ATLAS[atlas_key][self.name] = G.ASSET_ATLAS[self.atlas_key]
     end
 
-    for k, v in pairs(G.P_CENTER_POOLS.Tarot) do
-        SMODS.Consumable:take_ownership(v.key, { atlas = "Tarot" })
+    -- Call the `generate_colours` function of the appropriate type
```

Comment:
Again, create_colours is something I am wary of including in Steamodded itself. I'd like these functions to be cohesive, any specific functions can go in a separate mod.